### ISSUE TITLE
redis submodule cannot be imported due to circular import

### ISSUE THREAD

**Issue by ChrisGrosss** [WoLpH/portalocker#65]
Hi WoLpH,

thank you for developing and maintaining this package. I recently installed the version 2.3.0 via pip and figured out that the redis submodule did not properly install. It seems that the problem I've encountered can be traced back to line 10 in the redis submodule `from redis import client` which raises an `ImportError` due to a circular import. 

I tried the earliest version (2.1.0) in which the RedisLock has been implemented and also looked into the different branches of this project but this particular line of code seems to be in all of them, thus I do not know if I can circumvent this issue and still install and import the redis submodule or if I have missed something else.

Best regards,

Christian

**Comment by WoLpH** [comment]
Can you tell me the Python version you are using? It is probably related to that.

**Comment by WoLpH** [comment]
Can you try the develop branch to see if that helps?

**Comment by ChrisGrosss** [comment]
hi, thanks for the fast response.

I am currently running this on Python 3.8.5.

I downloaded and installed the develop branch. I saw that the redis submodule has been renamed to redis_lock. After running the example code `lock = portalocker.RedisLock('some_lock_channel_name')` I still got a TypeError as before `TypeError: 'NoneType' object is not callable`. When I imported the redis_lock submodule directly I observed a `ModuleNotFoundError` (`ModuleNotFoundError: No module named 'redis'`). Then I realised that redis is not the name of the original submodule that is part of the portalocker package but an independent package which needs to be installed first. 

Following, I looked into the setup.py and see that redis is stated as a required package but for an unknown reason it wasn't installed automatically during my previous attempts. After installing redis and reinstalling portalocker it seems to work without any problems.

Thank you for the fast response.

Best regards,
Christian
